README.md

```diff
@@ -2,365 +2,402 @@
+This is emitted when a key (and implied value) are removed from the global `meta` object.  If the key does not exist,
+then a [`warn`](#event-warn) event with the same signature will be emitted instead.
 
-##### cleanUpSIGINT
 
-*WARNING* Deprecated, you will have to opt-in and call cleanUpAll, where appropriate. AWS Lambda users have issues with listening on SIGTERM/SIGINT
+### Event: `'send'`
+* [`<Object>`][]
+    * `httpStatus` [`<String>`][] - The `status` property of the HTTP agent's response
+    * `firstLine` [`<String>`][] - The first log line in the buffer that was sent
+    * `lastLine` [`<String>`][] - The last log line in the buffer that was sent
+    * `totalLinesSent` [`<Number>`][] - The total number of lines in the sent buffer
+    * `totalLinesReady` [`<Number>`][] - The number of lines left to be sent (if queueing has happened)
+    * `bufferCount` [`<Number>`][] - The number of buffers left to be sent (if queueing has happened)
 
-##### ip
+This is emitted when a buffer is successfully sent to LogDNA.  Since a buffer can contain many log entries, this summarizes the activity.  
+In a high throughput system where `flushLimit` is exceeded and multiple buffers are waiting to be sent, information
+like `totalLinesReady` and `bufferCount` help illustrate how much work is left to be done.  Any buffers that have been queued will
+be sent one after another, ignoring any flush timer.
 
-* _**Optional**_
-* Type: `String`
-* Default: `''`
-* Values: `10.0.0.1`
 
-The default IP Address passed along with every log sent through this instance.
+### Event: `'warn'`
+* [`<Object>`][]
+    * `message` [`<String>`][] - The warn message.  Depends on context.
 
-##### level
+This is emitted when there is no log data provided to the `log` method, or when `removeMetaProperty` is called with an unrecognized key.  
+For those cases, additional properties (apart from `message`) are included:
 
-* _**Optional**_
-* Type: `String`
-* Default: `Info`
-* Values: `Debug`, `Trace`, `Info`, `Warn`, `Error`, `Fatal`, `YourCustomLevel`
-* Max Length: `32`
+**Warnings during `log()`**
 
-The default level passed along with every log sent through this instance.
+* `statement` (Any) - If `log()` was called with a `null` string, or an invalid data type, this will contain the given log statement
 
-##### mac
+**Warnings during `removeMetaProperty`**
 
-* _**Optional**_
-* Type: `String`
-* Default: `''`
-* Values: `C0:FF:EE:C0:FF:EE`
+* `key` [`<String>`][] - The key that was attempted to be removed, but did not exist
 
-The default MAC Address passed along with every log sent through this instance.
+## API
 
-##### max_length
+### `createLogger(key[, options])`
+
+* `key` [`<String>`][] - Your [ingestion key](https://docs.logdna.com/docs/ingestion-key)
+* `options` [`<Object>`][]
+    * `level` [`<String>`][] - Level to be used if not specified elsewhere. **Default:** `INFO`
```

Comment:
might be good to link to `level` like later in this doc

`[Level](#supported-log-levels) to be used ...`